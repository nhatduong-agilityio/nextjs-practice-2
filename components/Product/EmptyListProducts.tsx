import { Flex, Heading, Text } from '@chakra-ui/react'
import React, { memo } from 'react'

type EmptyListProductsProps = {
  searchQuery?: string
}

const EmptyListProducts = memo(
  ({ searchQuery }: EmptyListProductsProps): JSX.Element => (
    <Flex width='full' flexDir='column' alignItems='flex-start' justifyContent='center'>
      <Heading>No products found with {searchQuery}</Heading>
      <Text>Please try a different search term or category.</Text>
    </Flex>
  ),
)

EmptyListProducts.displayName = 'EmptyListProducts'

export default EmptyListProducts
