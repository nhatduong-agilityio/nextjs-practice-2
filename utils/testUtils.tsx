import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
