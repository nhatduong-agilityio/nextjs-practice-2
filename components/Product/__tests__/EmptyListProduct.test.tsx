// Utils
import { render } from '@utils/testUtils'

// Components
import EmptyListProducts from '../EmptyListProducts'

describe('EmptyListProducts render', () => {
  it('Should EmptyListProducts match Snapshot', () => {
    const { container } = render(<EmptyListProducts />)

    expect(container).toMatchSnapshot()
  })
})
