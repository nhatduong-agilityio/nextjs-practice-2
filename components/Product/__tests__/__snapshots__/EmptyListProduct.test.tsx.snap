// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyListProducts render Should EmptyListProducts match Snapshot 1`] = `
<div>
  <div
    class="css-6cgguy"
  >
    <h2
      class="chakra-heading css-1fhah9k"
    >
      No products found with 
    </h2>
    <p
      class="chakra-text css-ct6vz2"
    >
      Please try a different search term or category.
    </p>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
